metafields.go

```diff
@@ -0,0 +1,196 @@
+	WeightMetaFieldType MetafieldType = "weight"
+)
+
+// List types
+const (
+	// ListColorMetaFieldType is a list of hexadecimal color codes.
+	ListColorMetaFieldType MetafieldListType = "list.color"
+	// ListDateMetaFieldType is a list of dates in ISO 8601 format without presumed timezones.
+	ListDateMetaFieldType MetafieldListType = "list.date"
+	// ListDateTimeMetaFieldType is a list of dates and times in ISO 8601 format without presumed timezones.
+	ListDateTimeMetaFieldType MetafieldListType = "list.date_time"
+	// ListDimensionMetaFieldType is a list of values and a unit of length. Valid unit values: in, ft, yd, mm, cm, m.
+	ListDimensionMetaFieldType MetafieldListType = "list.date_time"
+	// ListFileReferenceMetaFieldType is a list references to a file on the online store. The default value is GenericFile. You can use validations to add other file types (for example, Image
+	ListFileReferenceMetaFieldType MetafieldListType = "list.file_reference"
+	// ListNumberIntegerMetaFieldType is a list of whole numbers in the range of +/-9,007,199,254,740,991.
+	ListNumberIntegerMetaFieldType MetafieldListType = "list.number_integer"
+	// ListNumberDecimalMetaFieldType is a list of numbers with decimal places in the range of +/-9999999999999.999999999.
+	ListNumberDecimalMetaFieldType MetafieldListType = "list.number_decimal"
+	// ListPageReferenceMetaFieldType is a list of references to pages on the online store.
+	ListPageReferenceMetaFieldType MetafieldListType = "list.page_reference"
+	// ListProductReferenceMetaFieldType is a list of product references.
+	ListProductReferenceMetaFieldType MetafieldListType = "list.product_reference"
+	// ListRatingMetaFieldType is a list of ratings measured on a specified scale. Validations are required for ratings.
+	ListRatingMetaFieldType MetafieldListType = "list.product_reference"
+	// ListSingleLineTextFieldMetaFieldType is a list of single-line text fields.
+	ListSingleLineTextFieldMetaFieldType MetafieldListType = "list.single_line_text_field"
+	// ListURLMetaFieldType is a list of URLs with one of the allowed schemes: https, http, mailto, sms, tel.
+	ListURLMetaFieldType MetafieldListType = "list.url"
+	// ListVariantReferenceMetaFieldType is a list of references to a product variant on the online store.
+	ListVariantReferenceMetaFieldType MetafieldListType = "list.variant_reference"
+	// ListVolumeMetaFieldType is a list of values and a unit of volume. Valid unit values: ml, cl, l, m3 (cubic meters), us_fl_oz, us_pt, us_qt, us_gal, imp_fl_oz, imp_pt, imp_qt, imp_gal.
+	ListVolumeMetaFieldType MetafieldListType = "list.volume"
+	// ListWeightMetaFieldType is a list of values and a unit of weight. Valid unit values: oz, lb, g, kg
+	ListWeightMetaFieldType MetafieldListType = "list.weight"
+)
+
+// Resources
+const (
+	// ArticleResource is the resource for metafields related to articles.
+	ArticleResource OwnerResource = "article"
```

Comment:
Why not just the below? Go will know that `ArticleResource` is a string based on the value `article`?

```suggestion
	ArticleResource = "article"
```

Reply:
Well, I did it because it's a type of its own and when it gets used in queries and stuff I referenced the type rather than a string. I guess I just threw it out there to see what you thought.

Reply:
Will change